lib/api/core/models/repositories/profileRepository.js

```diff
@@ -129,14 +128,13 @@ module.exports = function (kuzzle) {
       profile.roles = [ {_id: 'default'} ];
     }
 
-    Repository.prototype.hydrate.call(this, profile, data)
+    q(_.assignIn(profile, data))
```

Comment:
This line prevents changing the old code, but in the future I think it'll seem weird to anyone reading this.

I think it would be better to :
* get rid of the `result` variable entirely
* remove the `o` variable and use `profile` instead
* and to rewrite line 131 and the next 4-5 lines like that:

```js
_.assignIn(profile, data);
kuzzle.repositories.role.loadRoles(extractRoleIds(profile.roles))
  .then(...
```

And if you do follow these suggestions, it would be a good idea to remove the use of `q.defer()` and simply return the call to `loadRoles` to the caller.

What do you think?